### PR TITLE
move createGitOpsRepo mutation to go

### PR DESCRIPTION
**Pull request by salahalsaleh**

### CODE REVIEW

File: kotsadm/pkg/gitops/gitops.go [hunk 1]
```diff
@@ -134,10 +143,8 @@ func GetDownstreamGitOps(appID string, clusterID string) (*GitOpsConfig, error)
 		if splitKey[2] == "repoUri" {
 			if string(val) == repoURI {
 				// this is the provider we want
-				idx, err := strconv.ParseInt(splitKey[1], 10, 64)
-				if err != nil {
-				}
-				provider, publicKey, privateKey, repoURI, err := gitOpsConfigFromSecretData(idx, secret.Data)
+				idx, _ := strconv.ParseInt(splitKey[1], 10, 64)
```
Comment (divolgin) [inline kotsadm/pkg/gitops/gitops.go:146]: @marccampbell this here looks like it should fail on error and not ignore it.

File: kotsadm/pkg/handlers/gitops.go [hunk 2]
```diff
@@ -316,3 +325,34 @@ func GetGitOpsRepo(w http.ResponseWriter, r *http.Request) {
 
 	JSON(w, http.StatusOK, gitOpsConfig)
 }
+
+func CreateGitOps(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (divolgin) [inline kotsadm/pkg/handlers/gitops.go:330]: checkout the `handleOptionsRequest` function for this

File: kotsadm/pkg/gitops/gitops.go (hunk 1 above)
Comment (marccampbell) [reply kotsadm/pkg/gitops/gitops.go:146]: I don't disagree with your comment, but why me?  :)

Comment (salahalsaleh) [reply kotsadm/pkg/gitops/gitops.go:146]: @marccampbell probably because you wrote this block of code 🙂 . i'll update.

Comment (marccampbell) [reply kotsadm/pkg/gitops/gitops.go:146]: thanks @salahalsaleh.

Comment (divolgin) [reply kotsadm/pkg/gitops/gitops.go:146]: @marccampbell the red block looks like you wrote it.  but I only looked one commit back.

**Pull request merged**